### PR TITLE
Add end-to-end tests for type checking and variable property analysis

### PR DESCRIPTION
**Pull request by rkilar**

### CODE REVIEW

File: app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt [hunk 1]
```diff
@@ -1,5 +1,760 @@
...
+        assertErrorOfType(program, Diagnostic.ConditionalTypesMismatch::class)
+    }
+
+    fun assertUninitializedGlobalVariableError(program: String) {
+        assertErrorOfType(program, Diagnostic.UninitializedGlobalVariable::class)
+    }
+
+    fun assertConstantWithoutValueError(program: String) {
+        assertErrorOfType(program, Diagnostic.ConstantWithoutValue::class)
+    }
```
Comment (kamgal00) [inline app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt:23]: Those functions can be private, as well as assert functions in `VariablePropertiesAnalysisError`

File: app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt [hunk 2]
```diff
@@ -1,5 +1,760 @@
...
+
+    @Ignore
+    @Test
+    fun `test define variable with unit type`() {
+        assertInvalidTypeError("zm a: Nic;")
+        assertInvalidTypeError("wart a: Nic;")
+        assertInvalidTypeError("stała a: Nic;")
```
Comment (kamgal00) [inline app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt:30]: This test may also throw diagnostic `UninitializedGlobalVariable`. Maybe put those variables inside some functions?

**Review by kamgal00 — CHANGES_REQUESTED** [review]
Testy są wyczerpujące, wszystko wygląda ok :) wrzucam dwie małe uwagi

File: app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt [hunk 3]
```diff
@@ -1,5 +1,927 @@
...
+    private fun assertNonConstantExpressionError(program: String) {
+        assertErrorOfType(program, Diagnostic.NonConstantExpression::class)
+    }
+
+    @Ignore
+    @Test
+    fun `test define variable with unit type`() {
+        assertInvalidTypeError("czynność test() { zm a: Nic; }")
+        assertInvalidTypeError("czynność test() { wart a: Nic; }")
+        assertInvalidTypeError("czynność test() { stała a: Nic; }")
```
Comment (kamgal00) [inline app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt:35]: Sorry I didn't write it before, but I think variables and properties with type `Nic` are legal after all: they have only one possible value `nic`. Could we remove this test, as well as the next one?

**Comment by rkilar** [comment]
I added tests for `NonConstantExpression` errors. Right now we only consider integer and boolean literals to be constant so some cases are commented out. If we ever add more accurate checks, they can be added back.

**Review by kamgal00 — APPROVED** [review]
Wszystko jest ok, przy mergowaniu pamiętaj tylko żeby na razie zostawić Ignore przy `test wrong function call argument type`

**Pull request merged**